src/libmerklecpp/merklecpp.h

```diff
@@ -0,0 +1,1294 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#pragma once
+
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <list>
+#include <memory>
+#include <sstream>
+#include <stack>
+#include <vector>
+
+#ifdef WIN32
+#  include <stdlib.h>
+#  if BYTE_ORDER == LITTLE_ENDIAN
+#    define htobe32(X) _byteswap_ulong(X)
+#    define be32toh(X) _byteswap_ulong(X)
+#  else
+#    define htobe32(X) (X)
+#    define be32toh(X) (X)
+#  endif
+#  undef max
+#endif
+
+#ifdef WITH_TRACE
+#  include <iostream>
+#  define TRACE_HASH_SIZE 3
+#  define TRACE(X) \
+    { \
+      X; \
+      std::cout.flush(); \
+    };
+#else
+#  define TRACE(X)
+#endif
+namespace Merkle
```

Comment:
Our namespaces are all lower-case, I think we would prefer for this one to be too. This isn't enforced by the formatters, but it is something we stick to consistently.

Reply:
Sure, no problem.